lib/index.js

```diff
@@ -236,16 +267,130 @@ module.exports = class OtpRunner {
   }
 
   /**
-   * Builds a graph using OTP
+   * Recreate certain directories to make sure we're starting fresh
+   */
+  async clearFoldersAndFiles () {
+    try {
+      await fs.remove(this.getTargetFolder())
+      await fs.mkdirp(this.getTargetFolder())
+    } catch (e) {
+      this.log.error(e)
+      await this.fail('Failed to recreate graph router folder')
+    }
+  }
+
+  /**
+   * Returns the appropriate Graph.obj filename that OTP outputs after building
+   * a graph.
+   */
+  getGraphObjName () {
+    return this.isUsingOtp2x() ? 'graph.obj' : 'Graph.obj'
+  }
+
+  /**
+   * Downloads all needed files as specified in the manifest.
+   */
+  async downloadFiles () {
+    const downloadTasks = []
+
+    // add task to download OTP jar
+    downloadTasks.push({
+      dest: this.manifest.jarFile,
+      url: this.manifest.jarUrl
+    })
+
+    // add tasks to download files to the base folder
+    if (
+      this.manifest.baseFolderDownloads &&
+        this.manifest.baseFolderDownloads.length > 0
+    ) {
+      this.manifest.baseFolderDownloads.forEach(url => {
+        const splitUrl = url.split('/')
+        downloadTasks.push({
+          dest: path.join(
+            this.getTargetFolder(),
+            splitUrl[splitUrl.length - 1]
+          ),
+          url
+        })
+      })
+    }
+
+    // add tasks to download various GTFS files and then rename them according
+    // to OTP 2.x file naming requirements
+    if (
+      this.manifest.gtfsDownloads &&
+        this.manifest.gtfsDownloads.length > 0
+    ) {
+      this.manifest.gtfsDownloads.forEach(url => {
+        const splitUrl = url.split('/')
+        downloadTasks.push({
+          dest: path.join(
+            this.getTargetFolder(),
+            `gtfs-${splitUrl[splitUrl.length - 1]}`
+          ),
+          url
+        })
+      })
+    }
+
+    if (!this.manifest.buildGraph && this.manifest.runServer) {
+      // Manifest says to run the server without building a graph.
+      // Therefore, download a Graph.obj file.
+      downloadTasks.push({
+        dest: path.join(this.getTargetFolder(), this.getGraphObjName()),
+        url: this.manifest.graphObjUrl
+      })
+    }
+
+    // download files asynchronously
+    this.status.totalFilesToDownload = downloadTasks.length
+    await this.updateStatus(
+      `Downloading ${this.status.totalFilesToDownload} files...`,
+      10
+    )
+    await Promise.all(
+      downloadTasks.map(download => this.makeDownloadTask(download))
+    )
+  }
+
+  /**
+   * Make OTP build arguments according to the specified OTP version
+   */
+  makeBuildArgs () {
+    const args = getBaseOTPArgs(this.manifest.jarFile)
+    args.push('--build')
+    if (this.isUsingOtp2x()) {
+      args.push('--save')
+    }
+    args.push(this.getTargetFolder())
+    return args
+  }
+
+  /**
+   * Builds a graph using OTP if needed.
    */
   async buildGraph () {
+    // build graph if needed
```

Comment:
Seems like this comment should say `skip graph build if not needed`.

Reply:
Refactored in 2610ee9df6cb1878f067a5e2c1c277eed33836fe